### DISCUSSION TITLE
Web Demo to complete Javafx

### DISCUSSION THREAD

**Discussion by zepouet** [beehive-lab/TornadoVM#551] [Ideas/Proposals]
Hello,

Thanks a lot for the work and the existing demo about JavaFX. 
Is it possible to have a demo with SpringBoot or Quarkus ?
I have no idea if it is possible to integrate such deps.
My first tests fails and I think I don't understand exactly how deal tornado openjdk extension.
It could be great because it opens the demo to use FPGA in AWS (Xilink) or GPU and with websocket/SSR displays the results of computation. It could be great to migrate in web demo the JavaFX Raytracing.

Thanks a lot,
Best regards,

**Comment by jjfumero** [comment]
Hi @zepouet . Thanks for your interest. We haven't tried the integration of TornadoVM with neither SpringBoot nor Quarkus. I am nor familiar with these frameworks so I do not know if there will be issues with the integrations.  

Note that TornadoVM uses Java reflection for some classes regarding the API. However, the classes are known if you want to use the TornadoVM runtime, so for native images it should be possible. 

In a nutshell, TornadoVM is just another Java program. The only "special" thing is that it compiles Java bytecode at runtime by extending the Graal JIT compiler. For the rest, is just a bunch of JAR files with C++ shared libraries to dispatch the accelerated code (e.g., on GPUs and FPGAs).

**Comment by zepouet** [comment]
Thanks Juan. I will prepare a demo and commit publicly for my problem. Maybe I will find it by preparing the problem
